index.html

```diff
@@ -173,6 +173,44 @@ <h2>Calls for implementation and feedback</h2>
+
+A constraint is a restriction on the proposal, and may be used to withhold consensus for advancement. It takes the form of a well stated reason for why a proposal should not advance in it's present form, that is recorded by the committee. Well stated reasons should ideally illustrate the core issue of the constraint, this may be expressing a member industry position (such as a given design not working in the embedded space), a design concern, or a technical concern, and should provide an actionable plan whenever possible. The committee does not have an established concept of a rejected proposal.
+
+A block is the same as a constraint, however it is presented in a non-actionable way, or it cannot be addressed in the current stage and requires significant rework. This requires the champion to rework the proposal at an earlier stage in order to address the concern.
+
+In all possible cases, withholding consensus should take the form of a constraint. Blocks happen but should be avoided.
+
+<p>When a champion proposes a feature for advancement, a delegate may pose additional constraints and ask that the champion incorporate them into the proposal. The delegate should expect to spend time working with the champion to help flesh out the constraint and learn about other constraints that went into the proposal.
+
+<p>For features advancing from Stage 1 to Stage 2, a delegate may pose a constraint as necessary for advancement to Stage 3. In this situation, the delegate should expect to work with the champion and other delegates during Stage 2 to incorporate their constraint into the solution. In general, this is a good time to pose new constraints.
+
+<p>For features advancing to Stage 3, a delegate may ask that the champion work with them to incorporate a new constraint as a condition of advancement to Stage 3. In particular, given that consensus on Stage 3 means "the solution is complete", i.e., all open design issues havee beeen resolved including anticipated implementation and ecosystem compatibility issues, all TC39 participants should validate the design of proposals they care about before granting Stage 3 consensus.
+
+<p>Delegates should openly give feedback on proposals, especially when they have significant concerns, and especially for a proposal for stage advancement where the concern is relevant to the stage. Even if a Delegate does not intend to withhold consensus they should raise their concerns for general discussion.
+
+<p>A delegate may also request additional time to consider the proposal.
```

Comment:
I think this isn't completely unlimited (you can't just filibuster saying this forever), but setting a particular limit seems fraught, so let's just say it has to be reasonable.

```suggestion
<p>A delegate may also request a reasonable amount of additional time to consider the proposal. For example, proposal advancements are sometimes rejected for being posted fewer than 10 days before the start of the TC39 meeting (the "agenda deadline").
```

Reply:
Interesting. I had a different case in mind than the 10 days -- as thats highlighted in our agenda. What I had in mind is -- during discussion some topic comes up that a delegate hadn't considered before. They need to think about it a bit, but they don't want to block simply because they didn't consider it. In that case, they can request more time to think, and it either is resolved in the meeting, or in the very next meeting. Does this case seem like something we should write down?

Reply:
I think deferring to the next meeting is reasonable, and I wouldn't mind calling out that kind of scenario in this document. I don't know whether we should put an upper bound on it, but I can't really think of a scenario where delaying mutliple meetings to think is justified. We've had cases in the past where that was asked for, and pushed back against, though it's anyone's guess whether it theoretically would've been valid to assert an absolute right to block on that basis. I think that, if longer amounts of time for thinking about things are requested by delegates, then those delegates should give the champion some actionable request for how to facilitate the analysis (e.g., the champion could walk through the proposal with the delegate offline).